File: problema4.hs
-- Lista original
listaOriginal :: [String]
listaOriginal = ["rojo", "verde", "azul", "amarillo", "gris", "blanco", "negro"]

--Elementos a borrar
elementosABorrar :: [String]
elementosABorrar = ["amarillo", "cafe", "blanco"]

-- Función lambda para eliminar elementos
eliminarElementos :: [String] -> [String] -> [String]
eliminarElementos lista aBorrar = filter (\elemento -> not (elem elemento aBorrar)) lista


--Resultado 
listaResultante :: [String]
listaResultante = eliminarElementos listaOriginal elementosABorrar

main :: IO ()
main = do
    putStrLn "Lista original:"
    print listaOriginal
    putStrLn "\nElementos a borrar:"
    print elementosABorrar
    putStrLn "\nLista resultante:"
    print listaResultante

File: problema4.js
// Lista original
const listaOriginal = ['rojo', 'verde', 'azul', 'amarillo', 'gris', 'blanco', 'negro'];

// Elementos a borrar
const elementosABorrar = ['amarillo', 'cafe', 'blanco'];

// Función que elimina elementos usando lambda (función flecha)
const eliminarElementos = (lista, aBorrar) => {
    return lista.filter(elemento => !aBorrar.includes(elemento));
};

const listaResultante = eliminarElementos(listaOriginal, elementosABorrar)

console.log('Lista original:', listaOriginal);
console.log('Elementos a borrar:', elementosABorrar);
console.log('Lista resultante:', listaResultante);